core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/GuardrailsSupport.java

```diff
@@ -55,6 +55,30 @@ static UserMessage executeInputGuardrails(GuardrailService guardrailService, Use
         return um;
     }
 
+    static UserMessage executeInputGuardrails(GuardrailService guardrailService, UserMessage userMessage,
+            AiServiceMethodCreateInfo methodCreateInfo, ChatMemory chatMemory, AugmentationResult augmentationResult,
+            Map<String, Object> templateVariables, InvocationContext invocationContext) {
+        var um = userMessage;
+
+        if (guardrailService.hasInputGuardrails(methodCreateInfo)) {
+            var request = InputGuardrailRequest.builder()
+                    .userMessage(userMessage)
+                    .commonParams(
+                            GuardrailRequestParams.builder()
+                                    .chatMemory(chatMemory)
+                                    .augmentationResult(augmentationResult)
+                                    .userMessageTemplate(methodCreateInfo.getUserMessageTemplate())
+                                    .variables(templateVariables)
+                                    .invocationContext(invocationContext)
+                                    .build())
+                    .build();
+
+            um = guardrailService.executeGuardrails(methodCreateInfo, request);
+        }
+
+        return um;
+    }
+
```

Comment:
I just ran into this - why copy the entire method and not update all the call sites? This broke input guardrails support in certain use cases.

I'm fixing this in #1844